pkg/apiserver/rest/dualwriter.go

```diff
@@ -217,6 +232,26 @@ func SetDualWritingMode(
 		}
 	}
 
+	if (desiredMode == Mode3) && (currentMode == Mode2) {
```

Comment:
should we add a combination for `if (desiredMode == Mode3) && (currentMode == Mode2)` and ``if (desiredMode == Mode2) && (currentMode == Mode1)` ? Seems like it will be most of our use cases

Reply:
you mentioned two transitions `Mode2 -> Mode3` and `Mode1 -> Mode2`, however the first transition is already implemented (`Mode2 -> Mode3`). Perhaps were you referring to `Mode3 -> Mode2` (a downgrade)?

For `Mode1 -> Mode2`, it's alredy [implemented here](https://github.com/grafana/grafana/blob/5e638b4af0d25dd40da53ca5ebd0f6ebf3f58d23/pkg/apiserver/rest/dualwriter.go#L199)

Reply:
ah, I didnt see this part of the code 😄 but:
upgrade doesn't seem to do anything else rather than just setting the key? It doesn't seem do do any sort of sync. Is this intended?
Also, I think we also have a downgrade in [here](https://github.com/grafana/grafana/blob/5e638b4af0d25dd40da53ca5ebd0f6ebf3f58d23/pkg/apiserver/rest/dualwriter.go#L209)

Reply:
> upgrade doesn't seem to do anything else rather than just setting the key? It doesn't seem do do any sort of sync. Is this intended?

Actually, we have the `syncOk` verification on the [line just below](https://github.com/grafana/grafana/pull/89809/files#diff-e4fe089a41bba5a9ac7348794b35cb4c8edc29a0650f2e51bf8760f4d6492d36R239)